enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,114 @@ any user-defined primary network.
+| Subnets        | The subnet to use for the network across the cluster.<br/>E.g. 10.100.200.0/24.<br/>IPv6 (2001:DBB::/64) and dual-stack (192.168.100.0/24,2001:DBB::/64) subnets are supported.<br/>When omitted, the logical switch implementing the network only provides layer 2 communication, and users must configure IP addresses.<br/>Port security only prevents MAC spoofing if the subnets are omitted. | Yes      |
+| ExcludeSubnets | List of CIDRs and IP addresses.<br/>IP addresses are removed from the assignable IP address pool and are never passed to the pods.                                                                                                                                                                                                                                                                 | Yes      |
+| JoinSubnet     | Subnet used inside the OVN network topology. E.g. 100.65.0.0/24. IPv6 (fd99::/64) and dual-stack (100.65.0.0/24,fd99::/64). When omitted and network `role` is `primary` the following default CIDR is used `100.65.0.0/24`, IPv6 `fd99::/64`.                                                                                                                                                     | Yes      |
+
+The cluster scoped CRD should have the following additional field:
+
+| Field name | Description                                                                                          | optional |
+|------------|------------------------------------------------------------------------------------------------------|----------|
+| Selector   | Key-value list of labels used as a selector for which namespace the network should be available for. | No       |
+| Template   | The user defined network spec.                                                                       | No       |
+
+The selector type should be the standard `metav1.LabelSelector`.
+The template type should be the namespace scope CRD spec.
+
+> **Note:** The spec should be extended with care and strive to have minimal set of fields and provide abstraction for the NAD spec.
+
+##### status
+The CRD status should reflect the NAD state through conditions.
+For example, when the NAD its created the condition should be placed with status `True`.
+
+For cluster scoped networks, the condition should be true once all desired namespaces are provisioned with the corresponding NAD.
+
+The cluster scoped CRD status should reflect on which namespaces the network is available.
+
+##### Example - Namespace scoped network
+```yaml
+kind: UserDefinedNetwork
+metadata:
+  name: db-network
+  namespace: demo
+spec:
```

Comment:
we're missing the allow persistent ips knob here, aren't we ? 

We'd need it for cluster wide nets as well (but IIUC we'd get it "for free").

Reply:
The spec is changed to enable requesting networks with persistent IPs